gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -357,6 +354,23 @@ public void onCompleted() {
       }
     }
 
+    private void runWithRetries(Runnable block, SimpleResponseObserver responseObserver)
+        throws IOException {
+      for (int attemptedRetries = 0; ; attemptedRetries++) {
+        block.run();
```

Comment:
Will this update status of the `responseObserver` object?

Reply:
N/A anymore, but I think it should because the object in lambda is by reference.